src/crawlers/browser_crawler.js

```diff
@@ -360,12 +360,31 @@ export default class BrowserCrawler extends BasicCrawler {
      * @internal
      */
     async _handleRequestFunction(crawlingContext) {
-        const { id } = crawlingContext;
-        const page = await this.browserPool.newPage({ id });
-        this._enhanceCrawlingContextWithPageInfo(crawlingContext, page);
-
         const { request, session } = crawlingContext;
 
+        const newPageOptions = {
+            id: crawlingContext.id,
+        };
+
+        if (this.proxyConfiguration && this.launchContext && this.launchContext.useIncognitoPages) {
+            const { url: proxyUrl } = this.proxyConfiguration.newProxyInfo(session && session.id);
+            newPageOptions.proxyUrl = proxyUrl;
```

Comment:
I think that when using proxy per page, the `proxyInfo` object should be attached to the `crawlingContext` similarly to here. So that people can inspect which proxy they're using for the particular request.

https://github.com/apify/apify-js/blob/77626db26f7d9627116373596aabbc29c3a01da0/src/crawlers/cheerio_crawler.js#L540-L543

Otherwise, the browser launch `proxyInfo` and the `crawlingContext` proxy info would be different because the info would be taken from the browser.

https://github.com/apify/apify-js/blob/77626db26f7d9627116373596aabbc29c3a01da0/src/crawlers/browser_crawler.js#L511-L514

This is the part where the overwrites would happen:

https://github.com/apify/apify-js/blob/77626db26f7d9627116373596aabbc29c3a01da0/src/crawlers/browser_crawler.js#L414-L418

Reply:
Fixed. PTAL.

Reply:
I don't understand the change. The `proxyInfo` as well as the `session` would still get overwritten in the `_enhanceCrawlingContextWithPageInfo` function, no? We need to add some guard. Or we can move the logic inside the `_enhanceCrawlingContextWithPageInfo` function and either 

- create new `proxyInfo` and keep `session` if `useIncognitoPages` is `true`, or 
- take the existing `session` and `proxyInfo` from `browserController` when `useIncognitoPages` is `false`

Reply:
😅  PTAL again please